activemq_xml/tests/conftest.py

```diff
@@ -24,7 +24,26 @@ def dd_environment():
             os.mkdir(activemq_data_dir)
 
         with tarfile.open(fixture_archive, "r:gz") as f:
-            f.extractall(path=activemq_data_dir)
+            def is_within_directory(directory, target):
+                
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+            
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+                
+                return prefix == abs_directory
+            
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
```

Comment:
This syntax is not py2-compatible, so it would have to be:

```suggestion
            def safe_extract(tar, path=".", members=None, numeric_owner=False):
```